### DISCUSSION TITLE
502 Bad Gateway when calling getRelated in frontend

### DISCUSSION THREAD

**Discussion by aljazromih** [area17/twill#2454] [Q&A]
## Description

I get 502 Bad Gateway when calling `getRelated()` in a controller or blade on frontend. If I call it in a repeater blade (for dynamic select options) it works with no errors.

### Controller, blade view (502 Bad Gateway)
```php
$this->service = Service::forSlug($service)->published()->firstOrFail();
$this->doctors = $this->service->getRelated('doctors');
// dd($this->doctors);
// foreach ($this->service->getRelated('doctors') as $doctor) {
//     dd($doctor);
// }
```
```blade
{{-- @dd($service->getRelated('doctors')) --}}
@foreach ($service->getRelated('doctors') as $doctor)
  {{-- @dd($doctor) --}}
  <div>
    ...
  <div>
@endforeach
```

If I uncomment the `dd()`, `@dd()` (also in the foreach loop) above it works & returns a collection of the related models or single model within the foreach:
![This is an image](https://img001.prntscr.com/file/img001/lOvQlIqQSwmNOs-yrTIxIw.png)

### Sample in repeater blade view (works)
```php
foreach (\App\Models\Service::find($service)->getRelated('doctors') as $doctor) {
  $doctorOptions[] = [
    'value' => $doctor->id,
    'label' => $doctor->title
  ];
}
```

### Versions

Twill version: 2.12.4
Laravel version: 9.48.0
PHP version: 8.0

**Comment by ifox** [comment]
The 502 error is coming from your web server, but there must be an underlying error in your Laravel application for it to happen. Check your Laravel logs.

**Comment by aljazromih** [comment]
No error in log

> The 502 error is coming from your web server, but there must be an underlying error in your Laravel application for it to happen. Check your Laravel logs.

**Comment by ifox** [comment]
What about your web server/php-fpm logs?

**Comment by aljazromih** [comment]
> What about your web server/php-fpm logs?

Couldn't find any error.

**Comment by ifox** [comment]
Are you still reproducing this issue?
